services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -808,26 +808,28 @@ async def _get_tags_by_project(conn: SAConnection, project_id: str) -> List:
         )
         return [row.tag_id async for row in conn.execute(query)]
 
-    async def get_all_node_ids_from_workbenches(
-        self, project_uuid: str = None
-    ) -> Set[str]:
-        """Returns a set containing all the workbench node_ids from all projects
-
-        If a project_uuid is passed, only that project's workbench nodes will be included
-        """
-
-        if project_uuid is None:
-            query = "SELECT json_object_keys(projects.workbench) FROM projects"
-        else:
-            query = f"SELECT json_object_keys(projects.workbench) FROM projects WHERE projects.uuid = '{project_uuid}'"
-
+    async def node_id_exists(self, node_id: str) -> bool:
+        """Returns True if the node id exists in any of the available projects"""
         async with self.engine.acquire() as conn:
-            result = set()
-            query_result = await conn.execute(query)
-            async for row in query_result:
-                result.update(set(row.values()))
+            num_entries = await conn.scalar(
+                sa.select([func.count()])
+                .select_from(projects)
+                .where(projects.c.workbench.op("->>")(f"{node_id}") != None)
```

Comment:
`.op("->>")(f"{node_id}")` is this an exist check? How does it work?

Reply:
You can find the postgreql calls regarding json here:
https://www.postgresql.org/docs/12/functions-json.html
This returns the value of a key in a JSON dict for example. ```!= None``` means it asks whether it is ```IS NOT NULL``` in SQL, thus yes it checks for existence.